### PR TITLE
Publish additional metadata for recordkeeping and Centaur testability.  Closes #1006

### LINKED ISSUE #1006: Missing Functionality: JES Monitoring Script

### PR DESCRIPTION
**Pull request by mcovarr**

### CODE REVIEW

**Comment by geoffjentry** [comment]
@mcovarr could you describe what's being published?

**Comment by mcovarr** [comment]
For JES and Local backends:
- Workflow root
- Call root

JES backend only:
- Google project
- Execution bucket
- Endpoint URL
- GCS path to the monitoring log

File: supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesAsyncBackendJobExecutionActor.scala [hunk 1]
```diff
@@ -379,6 +379,12 @@ class JesAsyncBackendJobExecutionActor(override val jobDescriptor: BackendJobDes
       JesFileOutput(s"$MonitoringParamName-out", defaultMonitoringOutputPath.toString, Paths.get(JesMonitoringLogFile), workingDisk)
     }
 
+    tellMetadata(CallMetadataKeys.CallRoot, callRootPath)
+    tellMetadata("jes:googleProject", jesAttributes.project)
+    tellMetadata("jes:executionBucket", jesAttributes.executionBucket)
+    tellMetadata("jes:endpointUrl", jesAttributes.endpointUrl)
```
Comment (Horneth) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesAsyncBackendJobExecutionActor.scala]: Could be nice to have those in a `JesMetadataKeys` object too (like we have `CallMetadataKeys` and `WorkflowMetadataKeys`)

**Comment by Horneth** [comment]
👍

File: supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesAsyncBackendJobExecutionActor.scala (hunk 1 above)
Comment (mcovarr) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesAsyncBackendJobExecutionActor.scala]: good idea

**Comment by cjllanwarne** [comment]
Something something mumble mumble swagger.yaml...

But otherwise 👍

**Comment by mcovarr** [comment]
@cjllanwarne #1043 to address that point before we :shipit:

**Comment by geoffjentry** [comment]
👍

**Pull request merged**